pkg/smi/client_test.go

```diff
@@ -261,6 +346,15 @@ var _ = Describe("When listing TrafficTargets", func() {
 
 		_, err := fakeClientSet.smiTrafficTargetClientSet.AccessV1alpha3().TrafficTargets(testNamespaceName).Create(context.TODO(), trafficTarget, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
+
+		_, err = fakeClientSet.smiTrafficTargetClientSet.AccessV1alpha3().TrafficTargets("random-namespace").Create(context.TODO(), trafficTargetWithNamespaceMismatch, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = fakeClientSet.smiTrafficTargetClientSet.AccessV1alpha3().TrafficTargets(testNamespaceName).Create(context.TODO(), trafficTargetWithInvalidRuleKind, metav1.CreateOptions{})
```

Comment:
Am I correct that we expect this line to **not** create a new TrafficTarget because it is invalid. So then a few lines below when we list the traffic targets - the count will still be 1?
I wonder if a comment here would help disambiguate this.

Reply:
I updated these tests to make things clearer